File: strategy/python/strategies/credit_card_payment.py
from interfaces.payment_strategy import PaymentStrategy

class CreditCardPayment(PaymentStrategy):
    def __init__(self, card_number, cvv, expiry_date):
        self.card_number = card_number
        self.cvv = cvv
        self.expiry_date = expiry_date

    def pay(self, amount):
        print(f"Pagando ${amount} con Tarjeta de Crédito")
        print(f"Número de tarjeta: {self.card_number}")
        # Lógica de procesamiento de tarjeta de crédito

File: strategy/php/strategies/CreditCardPayment.php
<?php

class CreditCardPayment implements PaymentStrategy {
    private $cardNumber;
    private $cvv;
    private $expiryDate;

    public function __construct($cardNumber, $cvv, $expiryDate) {
        $this->cardNumber = $cardNumber;
        $this->cvv = $cvv;
        $this->expiryDate = $expiryDate;
    }

    public function pay($amount) {
        echo "Pagando $" . $amount . " con Tarjeta de Crédito\n";
        echo "Número de tarjeta: " . $this->cardNumber . "\n";
        // Lógica de procesamiento de tarjeta de crédito
    }
}